src/jdk.attach/linux/classes/sun/tools/attach/VirtualMachineImpl.java

```diff
@@ -77,6 +77,8 @@
         socket_path = socket_file.getPath();
         if (!socket_file.exists()) {
             File f = createAttachFile(pid, ns_pid);
+            // Keep a canonical version of File, to delete, in case target process ends and /proc link has gone:
+            File f2 = f.getCanonicalFile();
```

Comment:
Do we need to keep `f`? I think we can replace them to `createAttachFile(pid, ns_pid).getCanonicalFile()`.

Reply:
That's true, we don't really need them both.

Reply:
Builds and tests OK, thanks.